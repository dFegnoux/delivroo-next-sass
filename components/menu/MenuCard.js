import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import "./menuCard.scss";

class MenuCard extends PureComponent {
  static propTypes = {
    item: PropTypes.shape({
      id: PropTypes.string,
      title: PropTypes.string,
      description: PropTypes.string,
      picture: PropTypes.string,
      price: PropTypes.string,
      popular: PropTypes.bool
    }),
    updateCart: PropTypes.func.isRequired
  };

  static defaultProps = {
    item: {
      id: "Default id",
      title: "Default Menu item",
      description: "Default menu item description",
      picture: "//www.placekitten.com/300/300",
      price: "0.00",
      popular: false
    }
  };

  render() {
    const {
      item,
      item: { title, description, picture, price, popular },
      updateCart
    } = this.props;

    return (
      <div
        className="menuCard"
        onClick={() => {
          updateCart(item, "add");
        }}
      >
        <div className="content roundedItem">
          <div className="texts">
            <div className="cardTitle">{title}</div>
            <div className="description">{description.substr(0, 50)}</div>
            <div className="bottom">
              <span className="price">{price} â‚¬</span>
              {popular && <span className="popular">Populaire</span>}
            </div>
          </div>
          {picture && (
            <div className="pictureWrapper">
              <img src={picture} alt={title} />
            </div>
          )}
        </div>
      </div>
    );
  }
}

export default MenuCard;
