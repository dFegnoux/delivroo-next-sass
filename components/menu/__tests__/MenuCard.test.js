import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import MenuCard from "../MenuCard";

const props = {
  item: {
    id: "mocked id",
    title: "mocked title",
    description: "mocked description",
    picture: "mocked picture",
    price: "mocked price",
    popular: true
  },
  updateCart: jest.fn()
};
const container = shallow(<MenuCard {...props} />);

describe("MenuCard snapshots", () => {
  it("should render correctly with default props", () => {
    const tree = renderer.create(<MenuCard {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render correclty differents values of popular attribute", () => {
    const notPopular = renderer
      .create(<MenuCard {...props} item={{ ...props.item, popular: false }} />)
      .toJSON();
    expect(notPopular).toMatchSnapshot();

    const popular = renderer.create(<MenuCard {...props} />).toJSON();
    expect(popular).toMatchSnapshot();
  });

  it("should render correclty if item has picture or not", () => {
    const noPicture = renderer
      .create(<MenuCard {...props} item={{ ...props.item, picture: null }} />)
      .toJSON();
    expect(noPicture).toMatchSnapshot();

    const picture = renderer.create(<MenuCard {...props} />).toJSON();
    expect(picture).toMatchSnapshot();
  });
});

it("should handle click correctly", () => {
  container.simulate("click");
  expect(props.updateCart).toHaveBeenCalledWith(props.item, "add");
});
