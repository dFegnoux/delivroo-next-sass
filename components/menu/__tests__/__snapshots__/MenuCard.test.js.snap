// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuCard snapshots should render correclty differents values of popular attribute 1`] = `
<div
  className="menuCard"
  onClick={[Function]}
>
  <div
    className="content roundedItem"
  >
    <div
      className="texts"
    >
      <div
        className="cardTitle"
      >
        mocked title
      </div>
      <div
        className="description"
      >
        mocked description
      </div>
      <div
        className="bottom"
      >
        <span
          className="price"
        >
          mocked price
           €
        </span>
      </div>
    </div>
    <div
      className="pictureWrapper"
    >
      <img
        alt="mocked title"
        src="mocked picture"
      />
    </div>
  </div>
</div>
`;

exports[`MenuCard snapshots should render correclty differents values of popular attribute 2`] = `
<div
  className="menuCard"
  onClick={[Function]}
>
  <div
    className="content roundedItem"
  >
    <div
      className="texts"
    >
      <div
        className="cardTitle"
      >
        mocked title
      </div>
      <div
        className="description"
      >
        mocked description
      </div>
      <div
        className="bottom"
      >
        <span
          className="price"
        >
          mocked price
           €
        </span>
        <span
          className="popular"
        >
          Populaire
        </span>
      </div>
    </div>
    <div
      className="pictureWrapper"
    >
      <img
        alt="mocked title"
        src="mocked picture"
      />
    </div>
  </div>
</div>
`;

exports[`MenuCard snapshots should render correclty if item has picture or not 1`] = `
<div
  className="menuCard"
  onClick={[Function]}
>
  <div
    className="content roundedItem"
  >
    <div
      className="texts"
    >
      <div
        className="cardTitle"
      >
        mocked title
      </div>
      <div
        className="description"
      >
        mocked description
      </div>
      <div
        className="bottom"
      >
        <span
          className="price"
        >
          mocked price
           €
        </span>
        <span
          className="popular"
        >
          Populaire
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`MenuCard snapshots should render correclty if item has picture or not 2`] = `
<div
  className="menuCard"
  onClick={[Function]}
>
  <div
    className="content roundedItem"
  >
    <div
      className="texts"
    >
      <div
        className="cardTitle"
      >
        mocked title
      </div>
      <div
        className="description"
      >
        mocked description
      </div>
      <div
        className="bottom"
      >
        <span
          className="price"
        >
          mocked price
           €
        </span>
        <span
          className="popular"
        >
          Populaire
        </span>
      </div>
    </div>
    <div
      className="pictureWrapper"
    >
      <img
        alt="mocked title"
        src="mocked picture"
      />
    </div>
  </div>
</div>
`;

exports[`MenuCard snapshots should render correctly with default props 1`] = `
<div
  className="menuCard"
  onClick={[Function]}
>
  <div
    className="content roundedItem"
  >
    <div
      className="texts"
    >
      <div
        className="cardTitle"
      >
        mocked title
      </div>
      <div
        className="description"
      >
        mocked description
      </div>
      <div
        className="bottom"
      >
        <span
          className="price"
        >
          mocked price
           €
        </span>
        <span
          className="popular"
        >
          Populaire
        </span>
      </div>
    </div>
    <div
      className="pictureWrapper"
    >
      <img
        alt="mocked title"
        src="mocked picture"
      />
    </div>
  </div>
</div>
`;
